examples/audio/music-stream.lisp

```diff
@@ -0,0 +1,50 @@
+(in-package #:cl-user)
+(defpackage claylib/examples/music-stream
+  (:use :cl :claylib)
+  (:export :main))
+(in-package #:claylib/examples/music-stream)
+
+(defparameter *scene*
+  (make-scene ((music-ass (make-music-asset (claylib/examples:claylib-path
+                                             "examples/audio/resources/country.mp3"))))
+              ((title (make-text "MUSIC SHOULD BE PLAYING!" 255 150 :size 20 :color +lightgray+))
+               (timebar-bg (make-rectangle 200 200 400 12 +lightgray+))
+               (timebar-fill (make-rectangle 200 200 0 12 +maroon+))
+               (timebar-line (make-rectangle 200 200 400 12 +gray+ :filled nil))
+               (text1 (make-text "PRESS SPACE TO RESTART MUSIC"
+                                 215 250
+                                 :size 20
+                                 :color +lightgray+))
+               (text2 (make-text "PRESS P TO PAUSE/RESUME MUSIC"
+                                 208 280
+                                 :size 20
+                                 :color +lightgray+)))))
+
+(defun main ()
+  (with-window (:title "raylib [audio] example - music playing (streaming)"
+                :fps 30)
+    (with-audio-device
+      (with-scenes *scene* ()
```

Comment:
Have you by chance tried swapping these? I wonder if GC at the end of a scene might be throwing something off with the audio.

Reply:
On a somewhat related note, is there any reason we shouldn't just init the audio device during `with-window` by default? If the audio device is at all related that may just solve the problem and I think we'd probably want to do it anyway.

Reply:
Yeah, swapping them doesn't work because you have to have an audio context before loading music streams. The example will just run silently if you swap them. Intuitively (I could be very wrong), the GC shouldn't mess with the music stream in this case because it still only happens after we're ready to unload the streams. i.e. we exit the scope of with-scenes and do the GC after being done with our streams, then we exit the scope of with-audio-device which closes the audio device.